src/main/java/org/graylog/integrations/ipfix/codecs/IpfixCodec.java

```diff
@@ -82,18 +91,54 @@
     protected IpfixCodec(@Assisted Configuration configuration, IpfixAggregator ipfixAggregator) throws IOException {
         super(configuration);
         this.ipfixAggregator = ipfixAggregator;
-        // Standard IPFIX definition file is loaded from the classpath and is mandatory. We don't check for null or empty.
+
         final URL standardIPFixDefTemplate = Resources.getResource(IpfixCodec.class, IPFIX_STANDARD_DEFINITION);
-        final String ipFixCustomDefPath = (String) configuration.getSource().get(CK_IPFIX_DEFINITION_PATH);
-        if (ipFixCustomDefPath == null || ipFixCustomDefPath.trim().isEmpty()) {
+        final List<String> customDefFilePathList = configuration.getList(CK_IPFIX_DEFINITION_PATH);
+        List<URL> filePaths = new ArrayList<>();
+
+        if (customDefFilePathList == null || customDefFilePathList.isEmpty()) {
             infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate);
         } else {
-            final URL customIPFixDefURL = Paths.get(ipFixCustomDefPath).toUri().toURL();
-            infoElementDefs = new InformationElementDefinitions(standardIPFixDefTemplate, customIPFixDefURL);
+            errOnInvalidFilePath(customDefFilePathList);
+            filePaths.add(standardIPFixDefTemplate);
+            for (String filePath : customDefFilePathList) {
+                URL customDefURL = url(filePath.trim());
+                filePaths.add(customDefURL);
+            }
+            URL[] urls = array(filePaths);
+            infoElementDefs = new InformationElementDefinitions(urls);
```

Comment:
What do you think about changing the constructor signature of `InformationElementDefinitions` to take a `List` of URLs instead of an array of URLs? That way we can avoid the whole list to array conversion and remove the `array()` method again.

Reply:
@bernd - can we take a safe approach, to have two constructors, one to take a List<URL's>? Also, I like arrays . I have an hunch, that arrays might be light weight than List. Please advice.

Reply:
I would not use two constructors, that will complicate things. If you feel that arrays are better, then let's keep the arrays. :smiley: Arrays might be more lightweight, but in this case it's most probably not an issue.

Reply:
we are keeping arrays!!